samza-yarn/src/main/scala/org/apache/samza/job/yarn/SamzaContainerSecurityManager.scala

```diff
@@ -44,6 +48,39 @@ class SamzaContainerSecurityManager(config: Config, hadoopConfig: Configuration)
 
   private var lastRefreshTimestamp = 0L
 
+  def setApplicationAcl(ctx: ContainerLaunchContext) = {
+    val yarnConfig = new YarnConfig(config)
+    val viewAcl = yarnConfig.getYarnApplicationViewAcl
+    val modifyAcl = yarnConfig.getYarnApplicationModifyAcl
+    val acls: HashMap[ApplicationAccessType, String] = HashMap[ApplicationAccessType, String]()
+    if (viewAcl != null) {
+      acls += ApplicationAccessType.VIEW_APP -> viewAcl
+    }
+    if (modifyAcl != null) {
+      acls += ApplicationAccessType.MODIFY_APP -> modifyAcl
+    }
+    if (acls.nonEmpty) {
+      info("setting application acls in launch context: %s" format acls)
+      ctx.setApplicationACLs(acls.asJava)
+    }
+  }
+
+  def setupSecurityToken(fs: FileSystem, amContainer: ContainerLaunchContext): Unit = {
```

Comment:
Now that I see how it fits together, I actually think that exposing a getApplicationAcl() API on the YARNConfig class is the right call here.
`Map<ApplicationAccessType, String> getApplicationAcl()`

Both, ClientHelper and YarnClusterResourceManager should be able to invoke it since they have access to the `YarnConfig`. We wouldn't need to expose these methods on the `SamzaContainerSecurityManager` and can move the `setApplicationAcls` close to where it's actually required.

Reply:
So the reason to place these logic in SamzaContainerSecurityManager is so that we have a single place to handle security related logic. Had a initial discussion with @nickpan47 before I implemented this and he suggest SamzaContainerSecurityManager, that's how I ended up with this approach. I think your point is also valid. To me I don't feel either approach is significantly better. I would rather stick with what it is now.

So having YARNConfig to directly return Map<ApplicationAccessType, String> is an interesting idea. However, I think the Map<ApplicationAccessType, String> data structure is tightly coupled with the signature of ContainerLaunchContext. setApplicationACLs(). In the hadoop codes, view acl and modify acl are used independently. I would think it's simpler that the YARNConfig also treat them differently (getViewAcl and getModifyAcl as two independent methods and configs)

Reply:
Currently, the `SamzaSecurityManager` is scoped to the AMService and we shouldn't be instantiating multiple instances of it elsewhere. eg: With this, we're instantiating it in ClientHelper and YarnClusterResourceManager. 

Secondly, we should consider how the newly added methods fit in with the other methods on the SecurityManager. For eg: is it an expectation that we call `start` prior to invoking `setupSecurityToken`.  Until we have clear lifecycle, I'd prefer to avoid placing this logic inside the `SecurityManager` class - esp., if it entails something simple as setting some fields on the context.

@nickpan47 : What do you think?

Reply:
Yeah. the life cycle concern is a good point. I'm not familiar with scala and couldn't figure out how to do static method for a class. I have reverted my change on the security manager. And now do everything inside ClientHelper and YarnClusterResourceManager